From a6788d1da719ae006605b78d22f5a9f170b423af Mon Sep 17 00:00:00 2001
From: Tobias Junghans <tobydox@veyon.io>
Date: Wed, 27 May 2020 12:18:44 +0200
Subject: [PATCH] libvncserver: scale: cast to 64 bit before shifting

Since pixel_value is 64 bit the data type of the shift operand should
be 64 bit too to prevent integer overflows.
---
 libvncserver/scale.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libvncserver/scale.c b/libvncserver/scale.c
index a5cdf2baa..39ccb473d 100644
--- a/libvncserver/scale.c
+++ b/libvncserver/scale.c
@@ -210,7 +210,7 @@ void rfbScaledScreenUpdateRect(rfbScreenInfoPtr screen, rfbScreenInfoPtr ptr, in
              default:
                /* fixme: endianness problem? */
                for (z = 0; z < bytesPerPixel; z++)
-                 pixel_value += (srcptr2[z] << (8 * z));
+                 pixel_value += ((unsigned long)srcptr2[z] << (8 * z));
                 break;
               }
               /*
